src/Gui/View3DInventor.cpp

```diff
@@ -815,10 +818,11 @@ bool View3DInventor::setCamera(const char* pCamera)
             CamViewerO->aspectRatio      = ((SoOrthographicCamera *)Cam)->aspectRatio ;
             CamViewerO->height           = ((SoOrthographicCamera *)Cam)->height;
         } else {
+            Cam->unref();
             throw Base::TypeError("Camera type mismatch");
         }
     }
-
+    Cam->unref();

```

Comment:
It's known to be not good practice to free memory in all possible exit points of a function. A smarter way is to rely on C++'s RAII idiom which has two advantages:
* it keeps the code simple
* it even works if an unexpected exception is thrown

Reply:
I know that! But is there a scope lock/ handle/ smart ptr that can wrap an SoNode?

Reply:
For posterity:
CoinPtr works because Coin3D has already defined the intrusive_ptr methods in SoBase.h
```
// support for boost::intrusive_ptr<SoBase>
inline void intrusive_ptr_add_ref(SoBase * obj) { obj->ref(); }
inline void intrusive_ptr_release(SoBase * obj) { obj->unref(); }
```

Reply:
As said the CoinPtr class may do that. I am testing it right now...

Btw, I just realized two bugs in your code:
You call `Cam->ref();` directly after using the `SoDB::read` function and at the point Cam still could be a null pointer.
Then in the next line there is a check `if (!Cam){` and you call `Cam->unref();` which will cause a crash.

Reply:
Would you prefer for me to push again without (hopefully ) the stupidity and THEN delete my github account or maybe just delete my account, and you push directly to master?

Reply:
See https://github.com/FreeCAD/FreeCAD/commit/b49c523a541ce7ace520463e9e0b695ef6b20e5b

> delete my github account

You don't have to delete your account. Deleting the branch is enough :)